osu.Game/Graphics/Containers/SectionsContainer.cs

```diff
@@ -194,18 +212,27 @@ protected override void UpdateAfterChildren()
                 headerBackgroundContainer.Height = (ExpandableHeader?.LayoutSize.Y ?? 0) + (FixedHeader?.LayoutSize.Y ?? 0);
                 headerBackgroundContainer.Y = ExpandableHeader?.Y ?? 0;
 
-                float scrollOffset = FixedHeader?.LayoutSize.Y ?? 0;
+                // scroll offset is our fixed header height if we have it plus 20% of content height
+                // plus 5% to fix floating point errors and to not have a section instantly unselect when scrolling upwards
+                // but the 5% can't be bigger than our smallest section height, otherwise it won't get selected correctly
+                float sectionOrContent = Math.Min(smallestSection?.Height / 2.0f ?? 0, scrollContainer.DisplayableContent * 0.05f);
+                float scrollOffset = (FixedHeader?.LayoutSize.Y ?? 0) + scrollContainer.DisplayableContent * scroll_target_multiplier + sectionOrContent;
                 Func<T, float> diff = section => scrollContainer.GetChildPosInContent(section) - currentScroll - scrollOffset;
 
-                if (scrollContainer.IsScrolledToEnd())
+                if (Precision.AlmostBigger(0, scrollContainer.Current))
```

Comment:
`scrollContainer.IsScrolledToStart()` should really be a thing at this point (`ScorePanelList` has a similar local variant).